### PR TITLE
wine-nine-standalone: init at 0.9

### LINKED ISSUE #58701: package request: gallium nine standalone
## Issue description
Since neither the winetricks version nor the prebuilt version from [the official repo](https://github.com/iXit/wine-nine-standalone/releases) works with NixOS' package structure.

Mesa already supports the nine state tracker but there is no way to use it at the moment.

### PR DESCRIPTION
**Pull request by 9ary**
###### Description of changes

wine-nine-standalone provides the plumbing for Wine to use Gallium Nine, the D3D9 implementation in Mesa that is already shipped by NixOS. Upstream provides binary releases that are unlikely to work in a regular NixOS environment, so here it is nicely packaged.

Wine interfaces should be stable enough for a single build to be usable with all Wine variants and versions shipped by nixpkgs: upstream binary releases claim to work on a wide range of Wine versions.

Fixes #58701.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/wine-nine-standalone/setup_nine.sh
Comment (9ary) [inline pkgs/misc/wine-nine-standalone/setup_nine.sh:120]: One small problem I can see here is that in the 32 bit version of the wrapper, `$nine64` is actually going to point to the 32 bit version. This means that if you somehow end up with the 32 bit wrapper and wineWoW in your environment, you would install 32 bit Nine to the 64 bit path in the prefix. Pure 64 bit builds of Wine would probably fail to run ninewinecfg so it's not really a problem there.

Do we care?

File: pkgs/misc/wine-nine-standalone/default.nix
Comment (SuperSandro2000) [inline pkgs/misc/wine-nine-standalone/default.nix:13]:
```suggestion
  # Until and unless upstream issue https://github.com/iXit/wine-nine-standalone/issues/150
  # is resolved, this setup script is preferable.
  # When that gets fixed, WINEDLLPATH could be used instead.
```

File: pkgs/misc/wine-nine-standalone/nine.nix
Comment (SuperSandro2000) [inline pkgs/misc/wine-nine-standalone/nine.nix:54]: Please use clickable links

Comment (9ary) [reply pkgs/misc/wine-nine-standalone/nine.nix]: Thanks for the review, fixed both points.

File: pkgs/misc/wine-nine-standalone/default.nix
Comment (bb010g) [inline pkgs/misc/wine-nine-standalone/default.nix:17]: `${lib.escapeShellArg ./setup_nine.sh}`

File: pkgs/misc/wine-nine-standalone/default.nix
Comment (bb010g) [inline pkgs/misc/wine-nine-standalone/default.nix:16]: Double quote all shell parameter expansions.

**Review by bb010g — CHANGES_REQUESTED** [review]
All other review concerns were already addressed out-of-band.

**Review by bb010g — APPROVED** [review]

**Comment by 9ary** [comment]
Change of plans, I've submitted a couple of PRs upstream that address all my previous concerns and should eliminate the need for a setup script entirely. Things are now handled by a wrapper package that sets WINEDLLPATH. Let's at least wait until upstream changes are merged before merging this to avoid unnecessary breaking changes.

**Review by bb010g — APPROVED** [review]

File: pkgs/applications/emulators/wine-nine-standalone/nine.nix
Comment (SuperSandro2000) [inline pkgs/applications/emulators/wine-nine-standalone/nine.nix:39]:
```suggestion
    ./bootstrap.sh --distro nixpkgs
```
technically, not sure if that is some magic value

File: pkgs/applications/emulators/wine-nine-standalone/nine.nix
Comment (SuperSandro2000) [inline pkgs/applications/emulators/wine-nine-standalone/nine.nix:25]:
```suggestion
```

File: pkgs/applications/emulators/wine-nine-standalone/nine.nix
Comment (SuperSandro2000) [inline pkgs/applications/emulators/wine-nine-standalone/nine.nix:17]: If this is only used once, we might as well inline it

File: pkgs/applications/emulators/wine-nine-standalone/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/emulators/wine-nine-standalone/default.nix:20]: FYI we don't really need to quote $out but not important

File: pkgs/applications/emulators/wine-nine-standalone/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/emulators/wine-nine-standalone/default.nix:16]: Isn't this depending on stdenv.is32Bit?

Comment (9ary) [reply pkgs/applications/emulators/wine-nine-standalone/default.nix:16]: How so? The toplevel passes the following inputs to this package:
```
    nine32 = pkgsi686Linux.wine-nine-standalone-unwrapped;
    nine64 = wine-nine-standalone-unwrapped;
```
So on a 32 bit system, nine64 is actually the 32 bit build, and we just prefix WINEDLLPATH with the same value twice (and because of how `wrapProgram --prefix` works, the final value only contains it once).

For pure 32 bit wine packages on a 64 bit system, we do add the 64 bit version to the path, but that's no big deal, wine will just not use it.

File: pkgs/applications/emulators/wine-nine-standalone/default.nix
Comment (9ary) [reply pkgs/applications/emulators/wine-nine-standalone/default.nix:20]: Yeah, I know, @bb010g pushed me to defensively quote everything anyway and it doesn't hurt.

File: pkgs/applications/emulators/wine-nine-standalone/nine.nix
Comment (9ary) [reply pkgs/applications/emulators/wine-nine-standalone/nine.nix:39]: The upstream bootstrap sources /etc/os-release to extract the ID string. This is where that value comes from. Since /etc/os-release isn't available in a build environment I sent a patch upstream to take a command line argument instead, so it could have been anything really.
https://github.com/iXit/wine-nine-standalone/pull/153

File: pkgs/applications/emulators/wine-nine-standalone/default.nix
Comment (SuperSandro2000) [reply pkgs/applications/emulators/wine-nine-standalone/default.nix:16]: So we can make ``[[ -x "$out"/bin/wine ]]`` and ``[[ -x "$out"/bin/wine64 ]]`` dependend on stdenv.is32Bit and stdenv.is64Bit?

Comment (9ary) [reply pkgs/applications/emulators/wine-nine-standalone/default.nix:16]: The intent here is to do the right thing whether the input wine is 32 bit, wine64, or WoW.

**Comment by 9ary** [comment]
Closing this for now due to loss of interest (I don't need this anymore, and upstream is very unresponsive and unwilling to cooperate with packagers).

**Pull request closed without merging**
